internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,83 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	repositoryName   = "repo"
+)
+
+type checkoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func CloneCommandFor(s v1alpha1.GitSource) (string, error) {
+	cmd := &checkoutCmd{GitSource: s}
+	err := cmd.Validate()
+	if err != nil {
+		return "", err
+	}
+	return cmd.String(), nil
+}
+
+func (c *checkoutCmd) String() string {
+	var checkoutCommand string
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
+	var commit = c.Ref.Commit
+	var tag = c.Ref.Tag
+
+	if directory == "" {
+		directory = defaultDirectory
+	}
+
+	if commit != "" {
+		checkoutCommand = fmt.Sprintf("git clone %s %s && cd %s && git checkout %s && cp -r %s/* /manifests",
+			repository, repositoryName, repositoryName, commit, directory)
+		return checkoutCommand
+	}
+
+	if tag != "" {
+		checkoutCommand = fmt.Sprintf("git clone --depth 1 --branch %s %s %s && cd %s && git checkout tags/%s && cp -r %s/* /manifests",
+			tag, repository, repositoryName, repositoryName, tag, directory)
+		return checkoutCommand
+	}
+
+	checkoutCommand = fmt.Sprintf("git clone --depth 1 --branch %s %s %s && cd %s && git checkout %s && cp -r %s/* /manifests",
+		branch, repository, repositoryName, repositoryName, branch, directory)
+	return checkoutCommand
+}
+
+func (c *checkoutCmd) Validate() error {
```

Comment:
I think my preference would be to handle these kinds of validation checks in the CRD schema validation. Thoughts?